### ISSUE TITLE
Bad volume using get_dvh

### ISSUE THREAD

**Issue by Alxaline** [dicompyler/dicompyler-core#148]
Hello,

I used the version 0.5.6.
A big volume difference occurs with the output of `dvhcalc.get_dvh` which does not correspond to `CalculateStructureVolume`. The volume of `CalculateStructureVolume` seems to be alright by performing a check on another Software. So the DVH output of `get_dvh` are wrong. The issue seems similar to #26, but i did not understand how to get the right DVH value. How to resolve this point ?

Best regards,

**Comment by bastula** [comment]
Did you try checking the volume size with interpolation in `get_dvh`? If the structure is small relative to the RT Dose grid size, the voxellization process does not work well. That should result in an improvement in the volume calculation. You will need to set `interpolation_resolution` to a `float` if square pixels or a `tuple` if non-square. See https://dicompyler-core.readthedocs.io/en/latest/usage.html for more information.

**Comment by Alxaline** [comment]
Thank you for your reply. I already try to set to different `interpolation_resolution`. 
For a test case with `interpolation_resolution = 1 / 2 ** 5. (The increase don't change the value)
 ROI 1:
 - without `interpolation_resolution`: 1.8705000000000003 cm3
 - with `interpolation_resolution`: 1.873613012602903 cm3
- with `CalculateStructureVolume`: 6.420728688087663 cm3
ROI 2:
 - without `interpolation_resolution`:  47.43899999999999
 - with `interpolation_resolution`: 47.12509613641014 cm3
- with `CalculateStructureVolume`: 79.15991572455592 cm3

Dose image data information:
{'position': [-42.076000, -357.592000, -727.394000],
 'orientation': [1, 0, 0, 0, 1, 0],
 'pixelspacing': [1.000000, 1.000000],
 'rows': 99,
 'columns': 83,
 'samplesperpixel': 1,
 'photometricinterpretation': 'MONOCHROME2',
 'littlendian': True,
 'frames': 116}

**Comment by bastula** [comment]
Will need to do some more investigation. A quick workaround is to use the volume calculated from `CalculateStructureVolume` and manually set the volume on the `dvh` instance.

**Comment by Alxaline** [comment]
Thank you, pending a futur correction.
I think problems occurs when a part of the structure is outside of the dose map. For example the third example of Roi of my post above is a box of same dose map size.

**Comment by SergRuset** [comment]
> I think problems occurs when a part of the structure is outside of the dose map. For example the third example of Roi of my post above is a box of same dose map size.

Exactly, I've just checked this. This error from `dvhcalc.py` (`IndexError: too many indices for array`) occurs when one tries to calculate the DVH for the structure which is outside of the dosegrid (at least one voxel).

**Comment by bastula** [comment]
> This can be solved by using: `get_dvh(... calculate_full_volume=True ...)`

@SergRuset is correct. Does this resolve all the issues? I suppose I need to check if the structure is outside of the in-plane dose grid. The argument above I believe only solves the issues for structures that are out of the dose plane (i.e. superior or inferior).

**Comment by SergRuset** [comment]
Made some tests yesterday and, @bastula, you are absolutely right.
This solves only
> superior or inferior

**Comment by mbolt01** [comment]
For info in Oncentra brachy you can see the volume gets calculated differently in these cases too. In the contouring workspace the volume displayed is for the entire structure, however within the planning workspace the volume shown is only that within the dose calculation volume and so can be large differences as observed here.
